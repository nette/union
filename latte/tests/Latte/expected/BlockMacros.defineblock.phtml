<?php
%A%

class Template%a% extends Latte\Runtime\Template
{
	public $blocks = [
		'test' => 'blockTest',
		'testargs' => 'blockTestargs',
		'outer' => 'blockOuter',
	];

	public $blockTypes = [
		'test' => 'html',
		'testargs' => 'html',
		'outer' => 'html',
	];


	function main()
	{
		extract($this->params);
?>

<?php
		$var = 10;
?>

<?php
		if ($this->getParentName()) return get_defined_vars();
?>

a) <?php
		$this->renderBlock('test', ['var' => 20] + get_defined_vars(), 'html');
?>



b) <?php
		$this->renderBlock('testargs', [1] + get_defined_vars(), 'html');
?>



g) <?php
		$this->renderBlock('outer', get_defined_vars(), 'html');
?>


<?php
		$var1 = 'outer';
		?>h) <?php
		$this->renderBlock('testargs', get_defined_vars(), 'html');
?>


i) <?php
		$this->renderBlock('testargs', [null] + get_defined_vars(), 'html');
		return get_defined_vars();
	}


	function blockTest($_args)
	{
		extract($_args);
		?>	This is definition #<?php echo LR\Filters::escapeHtmlText($var) /* line 5 */ ?>

<?php
	}


	function blockTestargs($_args)
	{
		extract($this->params);
		list($var1, $var2) = $_args + [NULL, NULL, ];
		?>	Variables <?php echo LR\Filters::escapeHtmlText($var1) /* line 11 */ ?>, <?php echo LR\Filters::escapeHtmlText($var2) /* line 11 */ ?>, <?php
		echo LR\Filters::escapeHtmlText($hello) /* line 11 */ ?>

<?php
	}


	function blockOuter($_args)
	{
		extract($_args);
		$this->renderBlock('testargs', ['hello'] + get_defined_vars(), 'html');

	}

}
