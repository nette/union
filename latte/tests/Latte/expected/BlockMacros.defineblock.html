


a) 	This is definition #20



b) 	Variables 1, , world



g) 	Variables hello, , world


h) 	Variables , , world


i) 	Variables , , world
